File: src/main/java/net/siegemc/core/items/CustomItems/TestWand.java
package net.siegemc.core.items.CustomItems;

import net.siegemc.core.Core;
import net.siegemc.core.items.CreateItems.Rarity;
import net.siegemc.core.items.CreateItems.Stat;
import net.siegemc.core.items.CreateItems.WandItem;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import java.util.Arrays;

public class TestWand extends WandItem {
    public TestWand() {
        super("Test Wand", new ItemStack(Material.STICK));
        this.setManaRequired(10);
        this.setDamage(10000);
        this.setDamageRadius(5.5);
        this.setRange(120);
        this.setRarity(Rarity.DEBUG);
        this.getStats().put(Stat.STRENGTH, 100);
        this.setDescription(Arrays.asList("This is just a test wand used", "for testing wands"));
        Core.getItems().put("Test Wand", this);
    }
}

File: src/main/kotlin/net/siegemc/core/items/implemented/misc/wands/TestWand.kt
package net.siegemc.core.items.implemented.misc.wands

import net.siegemc.core.items.CustomItemUtils
import net.siegemc.core.items.enums.Rarity
import net.siegemc.core.items.recipes.recipes
import net.siegemc.core.items.types.misc.CustomWand
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class TestWand() : CustomWand(
    name = "Test Wand",
    customModelData = 1,
    description = listOf("A wand for testing"),
    levelRequirement = 0,
    material = Material.STICK,
    baseStats = CustomItemUtils.statMap(strength = 30.0),
    recipeList = recipes {

    },
    range = 15,
    red = 100,
    green = 100,
    blue = 100,
    damageRadius = 3.5
) {

    constructor(quality: Int): this() {
        this.quality = quality
        this.rarity = Rarity.getFromInt(quality)
    }

    constructor(item: ItemStack): this() {
        this.item = item
        deserialize()
    }

}